### PR TITLE
Feat/attachments

### PR DESCRIPTION
**Pull request by dannil76**
Nya endpoints

POST
/attachments

payload
`{
  "hashid": "yPVvdwDqmoN9JEd1op7401EGWQ8xjRL5",
  "attachmentId": "<unikt id som du behöver generera själv. referens till filen i viva>",
  "filename": "namn på filen",
  "fileBase64": "filen som base64 enkodad sträng"
}`

svar:
`{
    "ERRORCODE": null,
    "ERRORMESSAGE": null,
    "STATUS": "OK"
}`

GET
/attachments/<hash_id>/attachment/<attachment_id>

exempel på svar:
`{
    "DATA": null,
    "ERRORCODE": "1010",
    "ERRORMESSAGE": "Invalid key",
    "FILENAME": null,
    "STATUS": "Error"
}`

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def create_app():
             '/applications/<string:hash_id>/status',
         )
 
+        api.add_resource(
+            routes.Attachments,
+            '/attachments',
+            '/attachments/<string:hash_id>/attachment/<string:attachment_id>',
+        )
+
```
Comment (TeddyAndersson) [inline app/__init__.py:45]: Multiple endpoints defined on single resource, this behavior is not recommended from the python restful library docs. The library docs recommends that each endpoint has it's own resource.

```
add.api_resource(routes.AttachmentList, '/attachments')
add.api_resource(routes.Attachment, '/attachments/<string:hash_id>/attachment/<string:attachment_id>')
```

Also unclear if the retrieval of a singel attachment is necessary?

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]

**Comment by helsingborgsstad** [comment]
Task linked: [CU-ch1t1r VADA ska kunna skicka filer vidare till VIVA](https://app.clickup.com/t/ch1t1r)

File: app/__init__.py (hunk 1 above)
Comment (dannil76) [reply app/__init__.py]: Fixed

File: app/__init__.py [hunk 2]
```diff
@@ -40,6 +40,13 @@ def create_app():
...
             '/attachments',
+        )
+
+        api.add_resource(
+            routes.Attachment,
+            '/attachments',
+            '/attachments/<string:hash_id>',
+            '/attachments/<string:hash_id>/attachment',
             '/attachments/<string:hash_id>/attachment/<string:attachment_id>',
         )
```
Comment (TeddyAndersson) [inline app/__init__.py]: This still does not feel right. We should have one endpoint not four? 
I see this as four different endpoints

Already has an endpoint for: '/attachments'

Why all three? 
'/attachments/<string:hash_id>', // get all attachment for a user?
'/attachments/<string:hash_id>/attachment', // does not get anything?
'/attachments/<string:hash_id>/attachment/<string:attachment_id>', // get single attachment for a user? 

```
api.add_resource(
    routes.Attachments,
    '<string:hash_id>/attachments',
)
api.add_resource(
    routes.Attachment,
    '<string:hash_id>/attachments/<string:attachment_id>', or use query strings '/attachments/<string:attachment_id>?hash_id=<string:hash_id>'
)
```

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]

File: app/__init__.py (hunk 2 above)
Comment (TeddyAndersson) [reply app/__init__.py]: So my thought here is that everything relates to a user, as far as i know. With other words you cannot get or post attachments without a user connected to them? If this is the case then the {user_hash}/attachments/{attachment_id} feels more natural :)

Comment (dannil76) [reply app/__init__.py]: Tar ett snakk

Comment (TeddyAndersson) [reply app/__init__.py]: Ska vi ta tag i detta imorgon kanske?

Comment (dannil76) [reply app/__init__.py]: Anledningen till detta är att slippa få fula html-fel som svar om du går mot en endpoint som inte finns.
Ja, det finns säkert bättre sätt att lösa detta på som jag inte vet just nu.

Snakkes!

**Comment by dannil76** [comment]
Endpoint är nu inte konsekvent med hur resten av adapterns endpoint's är uppbyggda. I sak kanske denna nya struktur är mer rätt. Dock ser jag värde i att hålla endpoints genomgående konsekvent.

**Review by TeddyAndersson — APPROVED** [review]

**Pull request merged**